scalyr_agent/builtin_monitors/journald_monitor.py

```diff
@@ -0,0 +1,258 @@
+            try:
+                self._journal.seek_cursor( cursor )
+                entry = self._journal.get_next()
+
+                timestamp = entry.get( '__REALTIME_TIMESTAMP', None )
+                if timestamp:
+                    current_time = datetime.datetime.utcnow()
+                    delta = current_time - timestamp
+                    if delta.total_seconds() < self._staleness_threshold_secs:
+                        skip_to_end = False
+                    else:
+                        global_log.log( scalyr_logging.DEBUG_LEVEL_0, "Checkpoint is older than %d seconds, skipping to end" % self._staleness_threshold_secs )
+            except Exception, e:
+                global_log.warn( "Error loading checkpoint: %s. Skipping to end." % str(e) )
+
+        if skip_to_end:
+            # seek to the end of the log
+            # NOTE: we need to back up a single item, otherwise journald returns
+            # random entries
+            self._journal.seek_tail()
+            self._journal.get_previous()
+
+        # configure polling of the journal file
+        self._poll = select.poll()
+        mask = self._journal.get_events()
+        self._poll.register( self._journal, mask )
+
+    def _get_extra_fields( self, entry ):
+        """
+        Build a dict of key->values based on the fields available in the
+        passed in journal entry, and mapped to the keys in the 'journal_fields' config option.
+        """
+        result = {}
+
+        for key, value in self._extra_fields.iteritems():
+            if key in entry:
+                result[value] = str(entry[key])
+
+        return result
+
+    def _has_pending_entries(self):
+        """
+        Checks to see if there are any pending entries in the journal log
+        that are ready for processing
+        """
+
+        # do nothing if there is nothing ready to poll
+        if not self._poll.poll( self._poll_timeout ):
+            return False
+
+        # see if there are any entries to process
+        process = journal.NOP
+        try:
+            process = self._journal.process()
+        except Exception, e:
+            # early return if there was an error
+            global_log.warn( "Error processing journal entries: %s" % str(e),
+                             limit_once_per_x_secs=60, limit_key='journald-process-error' )
+
+            return False
+
+        if process == journal.NOP:
+            return False
+
+        return True
+
+    def _process_entries(self):
+        """
+        Processes all entries in the journal log.
+
+        Note: This function should only be called directly after a call to _has_pending_entries()
+        returns True.
+        """
+        # read all the entries
+        for entry in self._journal:
+            try:
+                msg = entry.get('MESSAGE', '')
+                extra = self._get_extra_fields( entry )
+                self._logger.emit_value('message', msg, extra_fields=extra)
```

Comment:
`message` is a reserved field in scalyr meant to hold the entire log line .  So, use something else here.. maybe `content` or `payload`?

Reply:
`details` appears to be used by other plugins, and I saw that during development and thought `message` is more suitable.  Now I now why that's the case :-)  Maybe switch it to `details`?

Reply:
Sure `details` sounds good to me.